src/Microsoft.ML/Runtime/Internal/Tools/CSharpApiGenerator.cs

```diff
@@ -177,6 +177,38 @@ public static string Capitalize(string s)
                 return char.ToUpperInvariant(s[0]) + s.Substring(1);
             }
 
+            public static string GetCharValue(char value)
+            {
+                switch (value)
+                {
+                    case '\t':
+                        return "\\t";
+                    case '\n':
+                        return "\\n";
+                    case '\r':
+                        return "\\r";
+                    case '\\':
+                        return "\\";
+                    case '\"':
+                        return "\"";
+                    case '\'':
+                        return "\\'";
+                    case '\0':
+                        return "\\0";
+                    case '\a':
+                        return "\\a";
+                    case '\b':
+                        return "\\b";
+                    case '\f':
+                        return "\\f";
+                    case '\v':
+                        return "\\v";
+                    default:
+                        Contracts.Assert(!Char.IsWhiteSpace(value));
```

Comment:
And a space, how about that?

Reply:
I'm not sure this is a useful assert.

---
In reply to: [188506120](https://github.com/dotnet/machinelearning/pull/142#discussion_r188506120) [](ancestors = 188506120)